tools/check_for_upgrade.sh

```diff
@@ -1,13 +1,17 @@
 #!/usr/bin/env zsh
 
+if [ -z "$ZSH_UPDATE" ]; then
```

Comment:
You can get rid of this check. The cache directory is a good enough location that it doesn't need to be customizable.